### ISSUE TITLE
BUGs in v0.74.3.5

### ISSUE THREAD

**Issue by lars18th** [cyd01/KiTTY#255]
Hy @cyd01 ,

Thank you for the last release v0.74.3.2. However, I see some bugs (perhaps the list is not complete):
1. When opening the editor (shortcut `editor`), if I select "File/ini file" now I see an error window with "Unable to load file".
2. I feel the file `docs/pages/cygtermd.md` is updated with and _older_ version. See this page for last modifications: https://github.com/cyd01/KiTTY/commit/232c3b29fd85f826bc40598677c823bc33cb7749#diff-1a9aebe19b12569225a1d5378255adf2863a82ce8f998fa8e1571a736329144b
3. The bug commented at #156 continues in my PC.
4. The parameter `-h` is not included: https://github.com/cyd01/KiTTY/issues/138#issuecomment-735971106

I only comment here for reference.
Regards.

**Comment by lars18th** [comment]
Hi @cyd01 ,

Please, almost restore the previous version of the file `docs/pages/cygtermd.md`, you overwrite it!

**Comment by lars18th** [comment]
Hi @cyd01 ,

Regarding the bugs in the last release v0.74.3.5:

- When using the new functionality "feat: new "New duplicated session..." menu item", if the current session is FROM a stored session AND using the PORTABLE mode, the New Session Window has EMPTY (default) values, instead of the values of the current session.

Please, revise it. For the rest I don't see any error.
Thank you!

**Comment by cyd01** [comment]
One more time I was not able to reproduce.  
I've started KiTTY in **portable** mode, load and start a **saved** session (either with start or open button). I've changed something (ex: change the background color to red) to be sure it is different from the default one, then choose "New duplicated session..." menu item.  
New session Window has the **current** session settings (i.e. the background color reflected my change).

**Comment by lars18th** [comment]
> One more time I was not able to reproduce.

OK, so I do this test: In a _clean_ empty directory I copied the kitty.exe binary and I repeat the test. And in this case... YES, it works!

So, I'll do more test to check why it doesn't work with my configuration... As _every_ time when I repeat the test it fails.

**Comment by lars18th** [comment]
Hi @cyd01 ,

I feel that I discovered the BUG... And I confirm that **IT DOESN'T WORK!**

Please, try this test to _reproduce_ the bug:
- In a computer without the registry `[HKEY_CURRENT_USER\Software\9bis.com\KiTTY]` create an empty directory and copy inside it the portable version of the KiTTY.
- Start the binary and close it.
- After that the Default config and INI are created.
- Re-Open and define a new session, changing the BACKGROUND color from default "0,0,0" to "100,100,100". Save the session and check it. All works? OK, please continue...
- Close the session and KiTTY. Edit the file `Default%20Settings` and replace the Background color to "50,50,50" (`Colour2\50,50,50\`), save the file.
- Then Reopen the KiTTY and load the saved session (with background color to 100), and select `New Duplicate Session...`. You then can see that the Background color is 100 instead of the default, and you think: then is working right... So you can open the new session with a different target and the "current" configuration (current background) is used in this new session. OK? Sorry, but NO!
- THIS IS NOT WORKING RIGHT! Please, open with an editor the file `Default%20Settings`, and see what happened: The **DEFAULT** background color is now 100 instead of the 50 !!!

What you're doing with this option is to RE-CREATE the _DEFAULT_ CONFIGURATION !!! But this is not the objective. The default configuration doesn't need to be touched in any sense. The objective is to _load_ the current configuration in the start window, and not to save the current configuration in the default configuration and then load this configuration. If you do in this sense this will fail, because first you touch the defualt configuration (bug) and second because if you can't save the default configuration (what happens with my configuration) then you use the default configuration instead of the current configuration (bug).

So, as identified the problem, please review the code.
Regards.

**Comment by cyd01** [comment]
Yes and no.  
"Background color is 100 instead of the default" => no. It is exactly what it is expected. The configuration box is open with the running configuration except for the hostname. That was you asked for. If you need the default setting you have the old "New session..." menu item.  
But you re right => the Default settings session must not be modified. I did not want to modify it. That's the bug. Normally I've planned to create a fake session starting with **__**. But it seems it was not done.

**Comment by cyd01** [comment]
As a quick workaround add these lines to your kitty.ini configuration file:

    [ConfigBox]
    defaultsettings=no

**Comment by cyd01** [comment]
Please try this [build](https://www.9bis.net/kitty/files/kitty-debug.exe).  
As I wrote before I did not change the behavior of the `New Duplicate Session...`.  
But now the `Default%20Settings` session file is not overwritten. The bug was in the initial creation function of this file at the very startup of **KiTTY**. The detection of the absence of the file was buggy ! So you can re-enable the `defaultsettings` option.

**Comment by lars18th** [comment]
Hi @cyd01 ,

> As a quick workaround add these lines to your kitty.ini configuration file:

I have this entry in my INI for years, and I suspect it's the cause because it doesn't work in my environment.

> As I wrote before I did not change the behavior of the `New Duplicate Session...`.
> But now the `Default%20Settings` session file is not overwritten. The bug was in the initial creation function of this file at the very startup of **KiTTY**. The detection of the absence of the file was buggy ! So you can re-enable the `defaultsettings` option.

Then it's nothing the check it _inside my configuration_ as I have the `defaultsettings=no`. So I'll check this new debug build in a standalone/clean directory.

> The configuration box is open with the running configuration except for the hostname. That was you asked for. If you need the default setting you have the old "New session..." menu item.

I know it! The _default configuration_ doesn't have _any_ sense when you select the new command `New Duplicate Session...`. And this command should use _only_ the _current_ configuration values. I'm not sure, but perhaps you're reading _some_ values from the default configuration?

Regards.

**Comment by lars18th** [comment]
Hi @cyd01 ,

I understand nothing! :fearful: 

I tested last debug build _inside_ my portable directory. And now when using the option `New Duplicate Session...` everytime a new window appears with "ici" (I feel is a french message indicating a debug info for you). So that's indicating that the function is called. Until here no problem.

However, I discoverd **two** different behaviours based on the _source_ of the configuration:
- If I open KiTTY and I put _all_ config values _manually_ (target address and background colour) then when I select the "new duplicate session" all is correct!
- But if I open KiTTY and I select a _saved_ session (with a background colour different from the default) then when I select the "new duplicate session" the values are only the _default_! So, no config parameters from the _current_ configuration! The background colour has not the values of the current session, but the default value.
 
And the last is true even if I change the background colour _after_ opening the saved session and before the duplicate session. Every time I use a saved session the command loads the default configuration and not the current one.

Please, try to reproduce in this way.
Regards.

**Comment by chrapuch** [comment]
Hi @cyd01 ,
I'm afraid the #197  (_Reconnect hangs on password entry_) is back. It's in v0.74.3.5, but not in v0.74.2.8

**Comment by cyd01** [comment]
@lars18th
I've got the same behavior: 
- with config manually entered
- with an already configured saved session

when I run `New Duplicate Session...` I have got a new config window with the settings of the running session. `Default Settings` session is not overwritten anymore.
I've tested it in Registry mode and of course in Portable mode.

**Comment by lars18th** [comment]
> @lars18th
> I've got the same behavior:
> I've tested it in Registry mode and of course in Portable mode.

I can't belive it! Ok. So I do more tests, as it doesn't work in my environment.

**Comment by lars18th** [comment]
Hi @cyd01 ,

First bug: The **last** debug-version, running it inside a _clean_ directory, and of course without anything in the registry (because I use only the portable mode)... it crash and doesn't create the default INI config.

Please, try it!

**Comment by cyd01** [comment]
You must create de ini file. The debug binary is always a full features one. So that you must specify that you enable the portable mode. Otherwise it will start in classic mode.

**Comment by cyd01** [comment]
Here is an example of Proxy definition file:

```
ProxyExcludeList\\
ProxyDNS\1\
ProxyLocalhost\0\
ProxyMethod\0\
ProxyHost\proxy\
ProxyPort\80\
ProxyUsername\\
ProxyPassword\\
ProxyTelnetCommand\connect%20%25host%20%25port%5Cn\
ProxyLogToTerm\1\
```

**Comment by lars18th** [comment]
Hi @cyd01 ,

> You must create de ini file. The debug binary is always a full features one. So that you must specify that you enable the portable mode. Otherwise it will start in classic mode.
> 
> Use `kitty.exe -defini` to create a configuration file template.

I discovered finally why you can't reproduce the bug. And now I'm sure you will can:

- Start with an empty directory, put KiTTY on it and generate the default configuration.
- Then open KiTTY and create a new session and save it. Remember to change something like for example the background colour.
- Exit KiTTY and _copy the saved session inside a subfolder of the Sessions directory_. Don't modify the session configuration, just copy it.
- Open KiTTY and open the saved session in the root folder. After select the `New Duplicate Session...` and check if the loaded values are the _current_ values. They are? Yes, so no bug...
- But, repeat it with the copied session inside the subfolder. What happens when selecting the `New Duplicate Session...` and check the loaded values. They are the _current_ values or the _default_ values? I'm sure you will get the _default_ values!

So, here the receipt to reproduce the bug.
And I see the bug because I don't have entries in the root sessions folder... all my sessions are ordered in subfolders.
Regards.

**Comment by lars18th** [comment]
> Here is an example of Proxy definition file:

And where to put it? Inside the INI file, or inside another different file?

**Comment by cyd01** [comment]
You got it !  
In portable mode it is necessary to explicitly specify the sub folder. Why ? Simply because you can have several sessions with same name in different subfolders. That's why there is a [-folder](https://www.9bis.net/kitty/#!pages/CommandLine.md) option available in portable mode.  
It is now specified at fake session startup too.

**Comment by lars18th** [comment]
Hi @cyd01 ,

I check it, and the problem is fixed in the last release v0.74.4.1. So I close this issue.

Thank you! :+1:
